python-avd/pyavd/_utils/format_string.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from collections.abc import Iterable
+from string import Formatter
+
+
+class AvdStringFormatter(Formatter):
+    """
+    Custom string formatter class to provide extra protection from malicious format strings and support for prefixes and suffixes per field.
+
+    The regular Python syntax is "{" [field_name] ["!" conversion] [":" format_spec] "}"
+    This class supports "{" [field_name] ["?"] ["<" prefix] [">" suffix] ["!" conversion] [":" format_spec] "}"
+
+    where
+        ? ::= The literal ? signals that the field is optional and will not be printed if the value is missing or None.
+        prefix ::= string including spaces which will be inserted before the field value. Most useful in combination with ?
+        suffix ::= string including spaces which will be inserted after the field value. Most useful in combination with ?
+        conversion ::= "!u" for "upper()" (The regular Python conversions "!r" | "!s" | "!a" have been removed).
```

Comment:
is this for security purposes?

Reply:
well !s is the default for format strings. We do not need ascii() or repr().